kazuki-core/src/main/kotlin/com/anaplan/engineering/kazuki/core/Sequence.kt

```diff
@@ -14,6 +14,8 @@ interface Sequence<out T> : List<T> {
 
     val inds: Set<nat1>
 
+    val tuples: Sequence<Tuple2<nat1, @UnsafeVariance T>>
```

Comment:
I'm not convinced by this name, as initially I read it as "tuple of elements from the sequence" and it took me a moment to figure out what was going on in the test. Given that all the other functions related to the position of elements are named around indices, I think something like `indexed` would be a more descriptive name (which surprisingly doesn't clash with an existing Kotlin function name).

Reply:
Resolved offline.

Reply:
discussed offline